include/marl/pool.h

```diff
@@ -366,11 +355,6 @@ class UnboundedPool : public Pool<T> {
     MARL_NO_EXPORT inline Storage(Allocator* allocator);
     MARL_NO_EXPORT inline ~Storage();
     MARL_NO_EXPORT inline void return_(Item*) override;
-    // We cannot copy this as the Item pointers would be shared and
```

Comment:
While I see you've moved this up to `Storage`, I think it would be generally safer if this remained here as well. Please revert.